elastic-agent-client.proto

```diff
@@ -101,26 +129,311 @@ message ActionResponse {
         // Action has failed.
         FAILED = 1;
     }
-    // Token that is used to unique identify the application to agent. When agent started this
+    // Token that is used to uniquely identify the application to agent. When agent started this
     // application it would have provided it this token.
     string token = 1;
     // Unique ID of the action.
     string id = 2;
     // Status of the action.
     Status status = 3;
-    // JSON encoded result for the action.
+    // JSON encoded result for the action (empty when diagnostic action response).
     bytes result = 4;
+    // Specific result for diagnostics action. (only used in V2)
+    repeated ActionDiagnosticUnitResult diagnostic = 5;
```

Comment:
Yeah type is not needed in the result, because we have the ID of the action. The type is taken by looking at the request. We would not care what the response says the type is, we would never trust that.